core/src/main/scala/com/raphtory/api/input/sources/JSONSource.scala

```diff
@@ -0,0 +1,70 @@
+package com.raphtory.api.input.sources
+
+import com.jayway.jsonpath.JsonPath
+import com.raphtory.api.input.Graph.assignID
+import com.raphtory.api.input._
+import com.raphtory.spouts.{FileSpout, ResourceSpout}
+
+/**
+ * * Specific format: NetworkX Node Link Data Format
+ * https://networkx.org/documentation/stable/reference/readwrite/generated/networkx.readwrite.json_graph.node_link_data.html
+ *
+ * This is a generic source to ingest data and build a graph in Raphtory, assuming the data is in Network X JSON node link format,
+ *
+ * You can configure source and target type and edge relationship type when calling this source in your Runner.
+ * You can also configure the key of your time value and edge type value if that exists in your data.
+ * Properties and keys of properties in your NetworkX graph will need to be added manually.
+ *
+ * @param spout : state where to ingest your data from (Mandatory field)
+ * @param sourceKey : state the key of the source ID in your JSON data (default = "source")
+ * @param sourceType : state what type of value the source is (default = "Name")
+ * @param targetKey : state the key of the target ID in your JSON data (default = "target")
+ * @param targetType : state what type of value the target is (default = "Name")
+ * @param edgeKey : state the key of the edge type/relationship in your JSON data (default = "Person to Person")
+ * @param timeKey : state the key of the timestamp in your JSON data (default = 1)
+ */
+
+class JSONSource(override val spout: Spout[String], sourceKey: String = "source", sourceType: String = "Name", targetKey: String = "target", targetType: String = "Name", edgeKey: Option[String] = None, timeKey: Option[String] = None) extends Source {
+  override type MessageType = String
+
+  private var source: String = _
+  private var target: String = _
+  private var _type: Option[String] = None
```

Comment:
Why the _ notation? The default Scala guideline is to escape with back-ticks: `type`

Reply:
sorry MY BAD!!